ci/request-reviews/get-reviewers.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# Get the code owners of the files changed by a PR,
+# suitable to be consumed by the API endpoint to request reviews:
+# https://docs.github.com/en/rest/pulls/review-requests?apiVersion=2022-11-28#request-reviewers-for-a-pull-request
+
+set -euo pipefail
+
+log() {
+    echo "$@" >&2
+}
+
+if (( "$#" < 5 )); then
+    log "Usage: $0 GIT_REPO BASE_REF HEAD_REF OWNERS_FILE PR_AUTHOR"
+    exit 1
+fi
+
+gitRepo=$1
+baseRef=$2
+headRef=$3
+ownersFile=$4
+prAuthor=$5
+
+tmp=$(mktemp -d)
+trap 'rm -rf "$tmp"' exit
+
+git -C "$gitRepo" diff --name-only --merge-base "$baseRef" "$headRef" > "$tmp/touched-files"
```

Comment:
This is not sound and apparently behaves differently to plain `git merge-base`, e.g. `git diff --name-only --merge-base origin/staging origin/haskell-updates` fails while you can compute `git merge-base origin/staging origin/haskell-updates`.

As a result, codeowners is currently send me an email on push as a service on haskell-updates.

Reply:
Hmm I cannot reproduce a `git diff` failure with the current branch heads, can you give me specific commits that caused that or a link to a failed run?

Reply:
Yes, sorry. E.g.: `git diff --name-only --merge-base df755a5e9c23fc1d9b5d801569e516889b28aec4 ea13a8bd5db362f54cce6a23178307217204e718` which should roughyl equate to [this run](https://github.com/NixOS/nixpkgs/actions/runs/12835006976/job/35793501215).